src/Features/CSharp/Portable/Snippets/CSharpConstructorSnippetProvider.cs

```diff
@@ -28,10 +26,9 @@ public CSharpConstructorSnippetProvider()
         {
         }
 
-        protected override async Task<bool> IsValidSnippetLocationAsync(Document document, int position, CancellationToken cancellationToken)
+        protected override bool IsValidSnippetLocation(in SnippetContext context, CancellationToken cancellationToken)
         {
-            var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
-            var syntaxContext = (CSharpSyntaxContext)document.GetRequiredLanguageService<ISyntaxContextService>().CreateContext(document, semanticModel, position, cancellationToken);
+            var syntaxContext = (CSharpSyntaxContext)context.SyntaxContext;
```

Comment:
would prefer this pattern in the previous file fwiw :)

Reply:
➡️ Resolved